src/server/routes/groups.js

```diff
@@ -0,0 +1,186 @@
+// endpoints for managing groups
+'use strict';
+
+const Logger = require('../logger'),
+    logger = new Logger('netsblox:api:Groups'),
+    // Storage = require('../src/server/storage/storage'),
+    // storage = new Storage(logger),
+    Users = require('../storage/users'),
+    assert = require('assert'),
+    Groups = require('../storage/groups');
+
+// assuming group names are unique (id)
+// TODO check if user has access to the group
+// TODO group-user association as admin or teacher
+
+module.exports = [
+    {
+        URL: 'groups',
+        Method: 'GET',
+        middleware: ['isLoggedIn'],
+        Handler: async function(req) {
+            // gets a list of groups
+            let owner = req.session.username;
```

Comment:
would this be the same situation as `getRawUserProjects` ? https://github.com/NetsBlox/NetsBlox/blob/master/src/server/storage/projects.js#L634

Reply:
kinda. It is less likely that a single user will have enough projects to cause any serious scalability issues but since we are listing all usernames on the server, this could have issues much more easily. In other words, the comparable call for projects would be listing all projects as an array (which will run into OOM on the current deployment due to the nodejs 1.5gb memory limit). This won't have the issue as easily as listing projects but much more easily than getting raw user projects. Does that make sense?

Reply:
what do you mean by 
> we are listing all usernames on the server

if it is that we are asking mongo to look through all usernames then yes it is doing that here. Then `getRawUserProjects` is also following a similar behavior as it will look through all projects correct?

note that `Groups.all(owner)` returns only the groups for that owner. I'm renaming it to `findAllUserGroups` to make it more clear and consistent with projects